lapack/gonum/dpstf2.go

```diff
@@ -0,0 +1,187 @@
+// Copyright ©2021 The Gonum Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style
+// license that can be found in the LICENSE file.
+
+package gonum
+
+import (
+	"math"
+
+	"gonum.org/v1/gonum/blas"
+	"gonum.org/v1/gonum/blas/blas64"
+)
+
+// Dpstf2 computes the Cholesky factorization with complete pivoting of an n×n
+// symmetric positive semidefinite matrix A.
+//
+// The factorization has the form
+//  Pᵀ * A * P = Uᵀ * U ,  if uplo = blas.Upper,
+//  Pᵀ * A * P = L  * Lᵀ,  if uplo = blas.Lower,
+// where U is an upper triangular matrix and L is lower triangular, and P is
+// stored as vector piv.
+//
+// Dpstf2 does not attempt to check that A is positive semidefinite.
+//
+// Dpstf2 is an internal routine. It is exported for testing purposes.
+func (Implementation) Dpstf2(uplo blas.Uplo, n int, a []float64, lda int, piv []int, tol float64, work []float64) (ok bool, rank int) {
```

Comment:
The order of the return values is surprising, all our others are `(some interestingvalue T, ok bool)`.